File: Data Structures & Algorithms/binary-tree-diameter/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public int diameterOfBinaryTree(TreeNode root) {
        int[] maxDiameter = new int[1];
        dfs(root, maxDiameter);
        return maxDiameter[0]; 
    }

    public int dfs(TreeNode root, int[] maxDiameter) {
        if(root == null) return 0;

        int left_height = dfs(root.left, maxDiameter);
        int right_height = dfs(root.right, maxDiameter);
        int diameter = left_height + right_height;
        maxDiameter[0] = Math.max(maxDiameter[0], diameter);
        return Math.max(left_height, right_height) + 1;
    }
}

File: Data Structures & Algorithms/binary-tree-right-side-view/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        if not root:
            return []
        
        queue = deque([root])
        res = []
        while len(queue) > 0:
            count = len(queue)
            last: TreeNode = None
            for i in range(count):
                last = queue.popleft()
                if last.left:
                    queue.append(last.left)
                if last.right:
                    queue.append(last.right)
            res.append(last.val)
        
        return res